hosting/couchdb/Dockerfile

```diff
@@ -1,14 +1,111 @@
-FROM couchdb:3.2.1
+# Modified from https://github.com/apache/couchdb-docker/blob/main/3.3.3/Dockerfile
```

Comment:
I thought we said 3.3.3 was compatible. Why do we need the modifications?

Reply:
The `couchdb:3.3.3` base image still uses bullseye. We need bookworm, so we've had to copy the contents of the 3.3.3 Dockerfile but base it on `node:20-slim` (chosen because it's bookworm and also the base image we use elsewhere, so introduces a little bit of consistency).